### PR TITLE
max3107 shell

### PR DESCRIPTION
**Pull request by kasjer**
This adds few commands for shell to manipulate registers of MAX3107.

Second commit moves MAX3107_0 instance settings to separate package
that will make it more clear what is instance and what is driver setting.

### CODE REVIEW

File: hw/drivers/uart/max3107/max3107_0/pkg.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+pkg.name: hw/drivers/uart/max3107/max3107_0
```
Comment (vrahane) [inline hw/drivers/uart/max3107/max3107_0/pkg.yml:20]: I don't think this should be an instance. It should just be `hw/drivers/uart/max3107`

File: hw/drivers/uart/max3107/max3107_0/pkg.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+pkg.description: Package adds single instance of MAX3107 device.
+pkg.author: "Apache Mynewt"
+pkg.keywords:
+    - uart
+
+pkg.init:
+    max3107_0_init: 'MYNEWT_VAL(MAX3107_0_SYSINIT_STAGE)'
```
Comment (vrahane) [inline hw/drivers/uart/max3107/max3107_0/pkg.yml:27]: this function is not going to be any different for different instances. So, I would just name it `max3107_init` and also the syscfg should be named `MAX3107_SYSINIT_STAGE`

File: hw/drivers/uart/max3107/max3107_0/syscfg.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    MAX3107_0_IRQ_PIN:
+        description: >
+            MAX3107 interrupt pin.
+        value: -1
+
+    MAX3107_0_LDOEN_PIN:
+        description: >
+            MAX3107 LDEON pin.
+        value: -1
+
```
Comment (vrahane) [inline hw/drivers/uart/max3107/max3107_0/syscfg.yml:121]: All above instances should not be _0_ I think, it's best to keep only one instance.

File: hw/drivers/uart/max3107/max3107_0/pkg.yml (hunk 1 above)
Comment (kasjer) [reply hw/drivers/uart/max3107/max3107_0/pkg.yml:20]: I disagree

File: hw/drivers/uart/max3107/max3107_0/pkg.yml (hunk 2 above)
Comment (kasjer) [reply hw/drivers/uart/max3107/max3107_0/pkg.yml:27]: I disagree

File: hw/drivers/uart/max3107/max3107_0/syscfg.yml (hunk 3 above)
Comment (kasjer) [reply hw/drivers/uart/max3107/max3107_0/syscfg.yml:121]: I disagree

**Review by vrahane — APPROVED** [review]
LGTM. Did not realize it was mostly re-organization.

**Pull request merged**